### PR TITLE
Add docs for setting up PhpStorm

### LINKED ISSUE #33: Add documentation for setting up PhpStorm
This should also include setting up proper JS hinting (see https://github.com/phpbb/documentation/pull/32#issuecomment-107150753).

### PR DESCRIPTION
**Pull request by VSEphpbb**
Closes #33

### CODE REVIEW

**Comment by VSEphpbb** [comment]
If anybody has anything to add or change 👍

File: development/development/phpstorm.rst [hunk 1]
```diff
@@ -0,0 +1,212 @@
+=========================
+Using PhpStorm With phpBB
+=========================
+
+PhpStorm is a leading IDE (Integrated Development Environment) and is the preferred development platform of the phpBB development team.
+
+This guide explains how to setup phpStorm for development with phpBB.
```
Comment (marc1706) [inline development/development/phpstorm.rst]: PhpStorm or phpStorm? :)

File: development/development/phpstorm.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+Create a New Project
```
Comment (marc1706) [inline development/development/phpstorm.rst]: Lower case `new`

File: development/development/phpstorm.rst [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+1. Visit the `phpBB repository <https://github.com/phpbb/phpbb>`_ and Fork it to your GitHub account.
+
+2. Clone your new phpBB fork to your computer.
+
+3. Open PhpStorm
+
+4. Choose "Create New Project from Existing Files" and follow the Wizard's steps.
+
+.. note:: Depending on your local environment, typically the best choise is "Source files are in a local directory, no web server is yet configured". You can always set up the web server later if want to use PhpStorm's server instead of a local WAMP/LAMP/MAMP stack.
```
Comment (marc1706) [inline development/development/phpstorm.rst]: `... the web server later if you want to`

File: development/development/phpstorm.rst [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+5. Select the phpBB project folder you cloned from GitHub and click "Project Root". Then click "Finish".
```
Comment (Crizz0) [inline development/development/phpstorm.rst]: No need to tell PhpStorm, where the "test"-dir is?

Comment (VSEphpbb) [inline development/development/phpstorm.rst]: Nope. Thats' what the configuration file is for.

File: development/development/phpstorm.rst [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+Code Style
+==========
+
+Setting up PhpStorm to honor phpBB's coding style guide will ensure you are always writing code that meets phpBB's requirements. These settings are configured in phpStorm's Options/Preferences, under **Editor > Code Style**.
```
Comment (marc1706) [inline development/development/phpstorm.rst]: Again, PhpStorm vs. phpStorm.

**Comment by Crizz0** [comment]
Maybe a section for the configuration of the VCS in PhpStorm? Or a advise to do not use it within PhpStorm? 

I don't know what the best would be. I just don't use this function.

File: development/development/phpstorm.rst (hunk 2 above)
Comment (VSEphpbb) [inline development/development/phpstorm.rst]: But it's a title, so title case?

**Comment by VSEphpbb** [comment]
VCS shouldn't require any setup. It will see the .git folder and know what to do based on that.

File: development/development/phpstorm.rst [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+Inspections
+===========
+
+One of the most powerful features of an IDE like PhpStorm is its ability to inspect your code and provide, in real time, errors and warnings. An IDE is fully aware of how phpBB's codebase is designed to work, and how all of its classes and functions interact with each other. Inspections can instantly show you, for example, if you are using a function incorrectly, calling non-existent methods, missing arguments, etc.
+
+The default inspection settings should work just fine. However there are a couple adjustments that might be desired:
+
+* If phpStorm is not connected to your database server, you may want to turn off **SQL > SQL Dialect Detection** and **SQL > No data sources configured warnings**.
+* phpBB uses fully qualified namespaces, so you can turn off this inspection warning **PHP > Code Style > Unnecessary fully qualified name**.
+* You may enable additional JavaScript inspections. Under **JavaScript > Code quality tools**, you can enable JSCS, JSHint, JSlint and/or ESLint.
```
Comment (marc1706) [inline development/development/phpstorm.rst]: We actually recommend JSHint and JSCS as we provide predefined inspection settings for these.

File: development/development/phpstorm.rst [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+PhpStorm is a leading IDE (Integrated Development Environment) and is the preferred development platform of the phpBB development team.
+
+This guide explains how to setup PhpStorm for development with phpBB.
+
+Create a New Project
+====================
+
+First you will need to get a copy of phpBB's development repository from GitHub.
+
+1. Visit the `phpBB repository <https://github.com/phpbb/phpbb>`_ and Fork it to your GitHub account.
```
Comment (paul999) [inline development/development/phpstorm.rst]: Lowercase Fork

File: development/development/phpstorm.rst [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+Tabs & Indents:
+***************
+
+Tabs should be used (not spaces). All tab and indent sizes should be set to 4 spaces. Also be sure "Keep indents on empty lines" is NOT checked.
+
+JSON, YAML
+##########
+
+Json and Yaml files uses 4 spaces over tabs, so tab character should be disabled and 4 spaces the indent size.
```
Comment (paul999) [inline development/development/phpstorm.rst]: JSON & YAML instead? To match title

**Comment by Derky** [comment]
I guess you need to link this file from the index.rst just like the coding_guidelines

**Comment by VSEphpbb** [comment]
what index.rst?

**Comment by Derky** [comment]
The contents are hidden because of Markdown I believe, so you need to look at the "Raw"-view: https://github.com/phpbb/documentation/blob/3.1.x/development/index.rst

**Comment by VSEphpbb** [comment]
Yeah I figured out. Updated.

**Pull request merged**